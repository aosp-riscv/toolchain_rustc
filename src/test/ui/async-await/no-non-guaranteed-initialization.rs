// compile-fail
// edition:2018
// compile-flags: --crate-type lib

<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
#![feature(async_await)]

async fn no_non_guaranteed_initialization(x: usize) -> usize {
    let y;
    if x > 5 {
        y = echo(10).await;
    }
    y
    //~^ use of possibly uninitialized variable: `y`
=======
async fn no_non_guaranteed_initialization(x: usize) -> usize {
    let y;
    if x > 5 {
        y = echo(10).await;
    }
    y
    //~^ use of possibly-uninitialized variable: `y`
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)
}

async fn echo(x: usize) -> usize { x + 1 }
