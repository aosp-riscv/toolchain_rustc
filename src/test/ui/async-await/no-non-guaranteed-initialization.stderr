<<<<<<< HEAD   (086005 Importing rustc-1.38.0)
error[E0381]: use of possibly uninitialized variable: `y`
  --> $DIR/no-non-guaranteed-initialization.rs:12:5
   |
LL |     y
   |     ^ use of possibly uninitialized `y`
=======
error[E0381]: use of possibly-uninitialized variable: `y`
  --> $DIR/no-non-guaranteed-initialization.rs:10:5
   |
LL |     y
   |     ^ use of possibly-uninitialized `y`
>>>>>>> BRANCH (8cd2c9 Importing rustc-1.39.0)

error: aborting due to previous error

For more information about this error, try `rustc --explain E0381`.
